flocker/node/agents/functional/test_ebs.py

```diff
@@ -65,6 +67,84 @@ class EBSBlockDeviceAPIInterfaceTests(
     """
     Interface adherence Tests for ``EBSBlockDeviceAPI``.
     """
+    def test_attach_foreign_instance_error(self):
+        """
+        Attempting to attach a volume to a non-local instance will
+        raise an ``AttachUnexpectedInstance`` error.
+        """
+        try:
+            config = get_blockdevice_config(ProviderType.aws)
+        except InvalidConfig as e:
+            raise SkipTest(str(e))
+        ec2_client = get_ec2_client_for_test(config)
+        dataset_id = uuid4()
+        volume = self.api.create_volume(dataset_id=dataset_id, size=ONE_GIB)
+
+        flocker_volume = ec2_client.connection.Volume(volume.blockdevice_id)
+        self.addCleanup(flocker_volume.delete)
```

Comment:
Seems better to do cleanup using the API object, in case it needs to do something else we don't know about.